leakcanary-android-core/src/test/java/leakcanary/internal/activity/screen/WrappedLineTest.kt

```diff
@@ -0,0 +1,152 @@
+"""
+
+    val wrappedString = LeakTraceWordWrapper.wrap(string, 80)
+
+    assertEquals("""
+┬───
+│ GC Root: System class
+│
+├─ leakcanary.internal.InternalAppWatcher class
+│    Leaking: NO (ExampleApplication↓ is not leaking and a class is never
+│    leaking)
+│    ↓ static InternalAppWatcher.application
+├─ com.example.leakcanary.ExampleApplication instance
+│    Leaking: NO (Application is a singleton)
+│    ExampleApplication does not wrap an activity context
+│    ↓ ExampleApplication.leakedViews
+│                         ~~~~~~~~~~~
+├─ java.util.ArrayList instance
+│    Leaking: UNKNOWN
+│    ↓ ArrayList.array
+│                ~~~~~
+├─ java.lang.Object[] array
+│    Leaking: UNKNOWN
+│    ↓ Object[].[0]
+│               ~~~
+├─ android.widget.TextView instance
+│    Leaking: YES (View.mContext references a destroyed activity)
+│    mContext instance of com.example.leakcanary.MainActivity with
+│    mDestroyed = true
+│    View#mParent is set
+│    View#mAttachInfo is null (view detached)
+│    View.mWindowAttachCount = 1
+│    ↓ TextView.mContext
+╰→ com.example.leakcanary.MainActivity instance
+​     Leaking: YES (ObjectWatcher was watching this because com.example.
+​     leakcanary.MainActivity received Activity#onDestroy() callback and
+​     Activity#mDestroyed is true)
+​     key = b3dd6589-560d-48dc-9fbb-ab8300e5752b
+​     watchDurationMillis = 5117
+​     retainedDurationMillis = 110
+""", wrappedString)
+  }
+
+  @Test
+  fun testMarker() {
+    val string = """
+│    ↓ UserProfile_Fragment.transaction↓ UserProfile_Fragment.transaction↓ UserProfile_Fragment.transaction
+│                                                                                               ~~~~~~~~~~~
+"""
+
+    val wrappedString = LeakTraceWordWrapper.wrap(string, 80)
+
+    assertEquals("""
+│    ↓ UserProfile_Fragment.transaction↓ UserProfile_Fragment.transaction↓
+│    UserProfile_Fragment.transaction
+│                         ~~~~~~~~~~~
+""", wrappedString)
+  }
+
+  @Test
+  fun testMarker2() {
+    val string = """
+├─ com.example.FooFooFooFooFooFooFoo instance
+│    Leaking: UNKNOWN
+│    ↓ FooFooFooFooFooFooFoo.barbarbarbarbarbarbarbar
+│                            ~~~~~~~~~~~~~~~~~~~~~
+"""
+
+    val wrappedString = LeakTraceWordWrapper.wrap(string, 30)
+
+    assertEquals("""
+├─ com.example.
+├─ FooFooFooFooFooFooFoo
+├─ instance
+│    Leaking: UNKNOWN
+│    ↓
+│    FooFooFooFooFooFooFoo.
+│    barbarbarbarbarbarbarbar
+│    ~~~~~~~~~~~~~~~~~~~~~
+""", wrappedString)
+  }
+
+  @Test
+  fun testSplitAndKeepByPeriodThenSplitBySpace() {
+    val string = "com.package is an application"
+
+    val splitted = string.splitAndKeep('.')
+
+    assertEquals(listOf("com.", "package is an application"), splitted)
+
+    val splitted2 = splitted.flatMap {
+      it.split(' ')
+    }
+
+    assertEquals(listOf("com.", "package", "is", "an", "application"), splitted2)
+  }
+
+  @Test
+  fun testSplitAndKeep() {
```

Comment:
nit: test names should never start with `test`, this is an artifact of Junit 3 which didn't have the `@Test` annotation (test methods had to start with `testX` to be detected back then.

Also, you can use backticks if you want spaces in test names for JVM tests.

Reply:
Good to know, thanks!